tests/perf_helper.py

```diff
@@ -0,0 +1,52 @@
+from typing import Any, Callable
+
+
+def run_perf_analytics(test_name: str, f: Callable[..., Any], *args, **kwargs) -> None:
+    """
+    Run performance analytics on a given function.
+
+    :param test_name: Name of the test.
+    :type test_name: str
+    :param f: Function to analyze performance for.
+    :type f: Callable[..., Any]
+    :param args: Positional arguments to pass to the function.
+    :param kwargs: Additional keyword arguments to pass to the function.
+    :type kwargs: dict
+    """
+    import cProfile
```

Comment:
Nice refactoring - really like it.

Just curious - why import stuff in the function? Why not just import at the top of module/file? Surely it is easier + cheaper.

Reply:
Laziness, bad copy pasta, will fix in next PR.

Reply:
Ahha - thought there is fancy features. Tyty.